core/src/main/java/com/novoda/merlin/service/MerlinService.java

```diff
@@ -67,10 +68,15 @@ private void disableConnectivityReceiver() {
 
     private void setReceiverState(int receiverState) {
         if (USE_COMPONENT_ENABLED_SETTING) {
-            getPackageManager().setComponentEnabledSetting(new ComponentName(this, ConnectivityReceiver.class), receiverState, PackageManager.DONT_KILL_APP);
+            getPackageManager().setComponentEnabledSetting(connectivityReceiverComponent(), receiverState, PackageManager.DONT_KILL_APP);
         }
     }
 
+    @VisibleForTesting
+    protected ComponentName connectivityReceiverComponent() {
+        return new ComponentName(this, ConnectivityReceiver.class);
```

Comment:
Ah my other comment didn't go through - my comment above is still valid (it won't matter which ctor we use).

These two tests fail for instance (with the same error):

```java
    @Test
    public void foo() throws Exception {
        ComponentName test = new ComponentName("foo", "bar");
        assertThat(test).isEqualTo(new ComponentName("foo", "bar"));
    }

//    java.lang.AssertionError: expected:
//    <'null (ComponentName@0)'>
//    but was:
//    <'null (ComponentName@0)'>
//    at com.novoda.merlin.service.MerlinServiceShould.foo(MerlinServiceShould.java:92)

    @Test
    public void bar() throws Exception {
        ComponentName test = new ComponentName("foo", "bar");
        assertThat(test).isEqualTo(new ComponentName("foasfo", "bsfsdfar"));
    }
```

We need to decide whether to:

- drop this branch and switch up to Robolectric 3 (also happy to do this)
- use androidTest and run these on connecto
- use Android Unit Tests and deal with it